### ISSUE TITLE
Process not stopped

### ISSUE THREAD

**Issue by shyim** [F1bonacc1/process-compose#33]
I am starting a PHP-FPM server using process-compose and after stopping process-compose its still active.

My config:

```
processes:
  phpfpm-web:
    command: /Users/shyim/Downloads/start.sh
```

My start.sh

```
#!/usr/bin/env bash

exec /nix/store/770sz2xzliv4xf3z7mgmhj1kbc3asy60-php-with-extensions-8.1.13/bin/php-fpm -F -y /nix/store/ib4r50ck8gnk3jviscz4d0rbr08x3qsh-phpfpm-web.conf -c /nix/store/13gh8w0v5ps7xj5lqwndld0in3bmzsc0-php.ini
```

The `-F` parameter disables deamonize of php-fpm, but for some reason its not stopped.

I started process compose and killed the process by own and it works.

**Comment by thenonameguy** [comment]
Hey @shyim! 

Assuming `process-compose` sends the shutdown signal correctly, you can always force the process to be force-killed:
```
phpfpm-web:
  command: /Users/shyim/Downloads/start.sh
  shutdown:
    signal: 9 # SIGKILL, do not trust handling of SIGTERM correctly
```
You are already using `exec` correctly, so the signal should go straight to `php-fpm`, can you try it with this?

**Comment by shyim** [comment]
Force killing does not help as the php-fpm childs still alive

**Comment by thenonameguy** [comment]
It seems this is a common problem with `php-fpm`:
https://github.com/usabilla/php-docker-template/issues/96

Not a process-compose issue.

**Comment by shyim** [comment]
I tried multiple signals non of them are working 🙈 

```
processes:
  phpfpm-web:
    command: /Users/shyim/Downloads/start.sh
    shutdown:
      signal: 3
```

kill -3 pid works

**Comment by shyim** [comment]
![image](https://user-images.githubusercontent.com/6224096/207297280-ac5cb568-cd00-4fa7-a117-b254347cb609.png)

```
❯ ps -ax | grep php-fpm
12668 ??         0:00.04 php-fpm: master process (/nix/store/ib4r50ck8gnk3jviscz4d0rbr08x3qsh-phpfpm-web.conf)
12669 ??         0:00.00 php-fpm: pool web
12670 ??         0:00.00 php-fpm: pool web
```
This is a different Pid somehow

**Comment by shyim** [comment]
Maybe related to my fish shell? 
![image](https://user-images.githubusercontent.com/6224096/207297669-f7aa0a71-aeb9-4492-9e58-3a208e598795.png)

**Comment by thenonameguy** [comment]
Good catch, might be. Try defining `command: sleep 50000` to remove `php-fpm` then from the equation.

**Comment by shyim** [comment]
![image](https://user-images.githubusercontent.com/6224096/207313516-b21753fa-e78d-4a5e-b8e6-e3fc9fca997e.png)

Seems like always executed through fish 😅

**Comment by F1bonacc1** [comment]
@shyim 
Does it work with bash?
You can change the used shell with the SHELL env variable.

**Comment by shyim** [comment]
Yep forcing with

```
SHELL=/bin/bash process-compose --config config.yaml
```

works 😅

**Comment by F1bonacc1** [comment]
As I see it there are a few things that I should improve:

1. I will add documentation regarding `fish` not being supported right now.
2. I can identify if `fish` is the value of `SHELL` and try to auto replace with bash.
3. Add the ability to configure the used shell in compose.yaml

@thenonameguy, @shyim do you see any other points for improvement?

**Comment by shyim** [comment]
Sounds good :)

**Comment by thenonameguy** [comment]
Sounds good to me, except point 2. 
For inspiration this is how `just` solves this (although they have their own DSL):
https://github.com/casey/just#shell

I think defaulting to `/usr/bin/env bash` on *nix-es and PATH `sh` on Windows makes sense in a similar fashion.
That would work on NixOS as well :tada: 
And would make the need for overriding every single compose.yaml file unnecessary for `fish` users.
The `compose.yaml` file should not be depending on the shell choice of the given developer IMO.

**Comment by F1bonacc1** [comment]
@thenonameguy I agree with your suggestion. If this was my thinking from the start, we wouldn't be having this conversation :)
I'll make bash the default.

**Comment by F1bonacc1** [comment]
@shyim
This issue should be solved in [v0.29.1](https://github.com/F1bonacc1/process-compose/releases/tag/v0.29.1)
 Let me know if you still have this issue.

**Comment by F1bonacc1** [comment]
@shyim, feel free to reopen this issue if it is not resolved for you.
